File: src/main/java/com/vjiki/music/service/CommentService.java
package com.vjiki.music.service;

import java.util.List;
import java.util.UUID;

import com.vjiki.music.dto.CommentReactionRequest;
import com.vjiki.music.dto.CommentRequest;
import com.vjiki.music.dto.CommentResponse;
import com.vjiki.music.pagination.CursorPageResponse;

public interface CommentService {

    CommentResponse addComment(CommentRequest request);

    List<CommentResponse> getCommentsByTrackId(UUID trackId, UUID userId);

    CursorPageResponse<CommentResponse> getCommentsPage(UUID trackId, UUID userId, int limit, String cursor);

    void addReaction(CommentReactionRequest request);

    void removeReaction(UUID commentId, UUID userId);

    void deleteComment(UUID commentId);
}

File: bin/main/com/vjiki/music/service/CommentService.kt
package com.vjiki.music.service

import com.vjiki.music.dto.CommentRequest
import com.vjiki.music.dto.CommentReactionRequest
import com.vjiki.music.dto.CommentResponse
import com.vjiki.music.pagination.CursorPageResponse
import java.util.UUID

interface CommentService {
    fun addComment(request: CommentRequest): CommentResponse
    fun getCommentsByTrackId(trackId: UUID, userId: UUID?): List<CommentResponse>
    fun getCommentsPage(trackId: UUID, userId: UUID?, limit: Int, cursor: String?): CursorPageResponse<CommentResponse>
    fun addReaction(request: CommentReactionRequest)
    fun removeReaction(commentId: UUID, userId: UUID)
    fun deleteComment(commentId: UUID)
}
